src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1875,6 +1875,32 @@ public void TestLongProcessIsWorking()
             Assert.True(p.HasExited);
         }
 
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [ActiveIssue(37054, TestPlatforms.OSX)]
+        [Fact]
+        public void LongProcessNamesAreSupported()
+        {
+            string programPath = GetProgramPath("sleep");
+
+            if (programPath == null)
+            {
+                return;
+            }
+
+            const string LongProcessName = "123456789012345678901234567890";
+            string sleepCommandPathFileName = Path.Combine(TestDirectory, LongProcessName);
+            File.Copy(programPath, sleepCommandPathFileName);
+
+            using (Process px = Process.Start(sleepCommandPathFileName, "600"))
+            {
+                Process[] runningProcesses = Process.GetProcesses();
+                Assert.Contains(runningProcesses, p => p.ProcessName == LongProcessName);
+
+                px.Kill();
+                px.WaitForExit();
```

Comment:
Why did you remove the try/finally you originally had here?

Reply:
I had copied it from the test that was in the other PR. I removed it because it should never throw.

Reply:
It will if the assert fails.

Reply:
Ah, yes. Let me add it back.